plugin.video.vimeo/resources/lib/kodi/search_history.py

```diff
@@ -0,0 +1,24 @@
+from time import time
+
+
+class SearchHistory:
+
+    filename = "search_history.json"
+
+    def __init__(self, settings, vfs):
+        self.settings = settings
+        self.size = int(self.settings.get("search.history.size"))
+        self.vfs = vfs
+
+    def get(self):
+        search = self.vfs.get_json_as_obj(self.filename)
```

Comment:
where does self.filename come from?

Reply:
It's a class variable defined at the top of the class.